automated/linux/docker-networking/docker-networking.sh

```diff
@@ -43,48 +50,83 @@ install_docker() {
-DOCKER_INSPECT=$(docker network inspect bridge)
-exit_on_fail "docker-network-inspect" "${skip_list}"
-
-echo "$DOCKER_INSPECT" | jq '.[0]["Containers"][]'
-IP_ADDR=$(echo "$DOCKER_INSPECT" | jq '.[0]["Containers"][] | select(.Name=="ping_test_container") | .IPv4Address | split("/")[0]')
-echo "${IP_ADDR}"
-if [ -n "$IP_ADDR" ]; then
-    report_pass "docker-network-bridge"
-    eval "ping -c4 $IP_ADDR"
-    check_return "ping-container-test"
-else
-    report_fail "docker-network-bridge"
-    report_skip "ping-container-test"
-fi
+for NETWORK in ${NETWORK_TYPE}; do
+    # check if bridge network is present
+    skip_list=$(remove_one_from_skiplist "${skip_list}")
+    docker network ls -f name="${NETWORK}" | grep "${NETWORK}"
+    exit_on_fail "docker-network-list" "${skip_list}"
+
+    # start simple alpine container
+    skip_list=$(remove_one_from_skiplist "${skip_list}")
+    docker run --name ping_test_container --network "${NETWORK}" --rm -d "${IMAGE}" /bin/sleep 90
+    exit_on_fail "docker-start-container" "${skip_list}"
+
+    # container should join NETWORK network
+    skip_list=$(remove_one_from_skiplist "${skip_list}")
+    DOCKER_INSPECT=$(docker network inspect "${NETWORK}")
+    exit_on_fail "docker-network-inspect-${NETWORK}" "${skip_list}"
 
-docker kill ping_test_container
-check_return "docker-kill-container"
+    skip_list=$(remove_one_from_skiplist "${skip_list}")
+    if [ "${NETWORK}" = "bridge" ]; then
+        echo "$DOCKER_INSPECT" | jq '.[0]["Containers"][]'
+        IP_ADDR=$(echo "$DOCKER_INSPECT" | jq '.[0]["Containers"][] | select(.Name=="ping_test_container") | .IPv4Address | split("/")[0]')
+        echo "${IP_ADDR}"
+        if [ -n "$IP_ADDR" ]; then
+            report_pass "docker-network-${NETWORK}"
+            eval "ping -c4 $IP_ADDR"
+            skip_list=$(remove_one_from_skiplist "${skip_list}")
+            check_return "ping-container-test-${NETWORK}"
+        else
+            report_fail "docker-network-${NETWORK}"
+            skip_list=$(remove_one_from_skiplist "${skip_list}")
+            report_skip "ping-container-test-${NETWORK}"
+        fi
+    else
+        report_pass "docker-network-${NETWORK}"
+        skip_list=$(remove_one_from_skiplist "${skip_list}")
+        report_skip "ping-container-test-${NETWORK}"
+    fi
+    skip_list=$(remove_one_from_skiplist "${skip_list}")
+    docker kill ping_test_container
+    check_return "docker-kill-container-${NETWORK}"
+
+    skip_list=$(remove_one_from_skiplist "${skip_list}")
+    if [ -n "${HOST_IP}" ]; then
+        if [ "${NETWORK}" = none ]; then
+            # ping should fail with disabled networking
+            if ! docker run --name ping_localhost_host_network --network "${NETWORK}" --rm "${IMAGE}" ping -4 -c 4 "${HOST_IP}"; then
+                report_pass "docker-ping-host-network-${NETWORK}"
+            else
+                report_fail "docker-ping-host-network-${NETWORK}"
+            fi
```

Comment:
can this if else statement be written to check_return instead?

Reply:
I'm expecting this to fail (ping shouldn't work with `none` networking). check_return will record fail where it should record pass.

Reply:
Could we do something like this in the check_return function?

`$ git df
diff --git a/automated/lib/sh-test-lib b/automated/lib/sh-test-lib
index bce0e716efa9..2dc14475cc5e 100755
--- a/automated/lib/sh-test-lib
+++ b/automated/lib/sh-test-lib
@@ -105,12 +105,13 @@ check_return() {
     [ "$#" -ne 1 ] && error_msg "Usage: check_return test_case"
     # shellcheck disable=SC2039
     local test_case="$1"
+    local xfail="${2:-}"
 
-    if [ "${exit_code}" -ne 0 ]; then
-        echo "${test_case} fail" | tee -a "${RESULT_FILE}"
+    if [ "${exit_code}" -ne 0 && -z "${xfail}" ]; then
+        report_fail "$1"
         return "${exit_code}"
     else
-        echo "${test_case} pass" | tee -a "${RESULT_FILE}"
+        report_pass "$1"
         return 0
     fi
 }
`

Reply:
Added to sh-test-lib.